core/modules/system/system.module

```diff
@@ -2948,10 +2948,12 @@ function _system_rebuild_module_data() {
           }
         }
         if ($git_version) {
-          $git_version = preg_replace('/^' . preg_quote(BACKDROP_CORE_COMPATIBILITY, '/') . '-/', '', $git_version);
           $modules[$key]->info['version'] = $git_version . '-dev';
         }
       }
+      else {
+        $modules[$key]->info['version'] = BACKDROP_CORE_COMPATIBILITY . '-unknown';
```

Comment:
Assuming BACKDROP_CORE_COMPATIBILITY seems is a bit problematic to me.

I tested locally with a dummy module, that didn't even have "backdrop = 1.x" set in its info file, and the version showed up as "1.x-unknown".

Reply:
As I know BACKDROP_CORE_COMPATIBILITY is set in the core, not by a contributed module. Also it effects only the first part, the "-unknown" part is from your own suggestion in https://github.com/backdrop/backdrop-issues/issues/5002#issuecomment-800103642 That simply means your dummy module was not downloaded from github.

Reply:
In other words, when Backdrop becomes 2.x all contrib module first part also become 2.x, so BACKDROP_CORE_COMPATIBILITY is exactly what we want there.

Reply:
Sure, but...

If I adapt the dummy module info file like that: `backdrop = 2.x` I still end up with `1.x-unknown`. I do find this problematic.

Reply:
But then the modules should follow the core version. Remember per your suggestion it the version should have format `core-major.minor.patch`, so until Backdrop core goes 2.x all modules should ideally declare `backdrop = 1.x`. Otherwise - by following what modules declared - we will have real mess on the Modules page.

Reply:
You seem to miss my point. :wink: Your code *assumes* that this module is for 1.x without any evidence. So the 1.x may be contradictory to the actual info file - which I wouldn't recommend.

Reply:
> I got that part, but I thought that modules can not be for 2.x or higher before the core goes beyond 1.x....

It's not that I insist. But I suspect you did *not* yet get what I mean. :wink: I'll give it another try:

Yes, the module can't get installed, it shows a red crossed button instead of a checkbox in the listing. But that's not my point. If adding "1.x" to the version string, you claim that this module is for core 1.x - but that's only an assumption and can be wrong. You use the constant from core (BACKDROP_CORE_COMPATIBILITY), which has nothing to do with the actual value from the module's info file. Did I explain it better now?

The check for core compatibility happens somewhere else. Our goal is only to improve the version display.

Reply:
Ok, I see what you mean. So do you want me to change the code, so the `core` part is derived from the module's info?

Reply:
> So do you want me to change the code, so the core part is derived from the module's info?

That's your decision. My request was to avoid misleading info. If it's easy to achieve - sure, why not. If it's complicated, you can stick with just the word "unknown" - IMO that's already an improvement to the current display (nothing at all).

Reply:
It's not difficult as info about a module is already sitting there in variables, however taking into account the contrib modules cab write after `backdrop =` anything, I think your idea to leave just 'unknown' will provide a cleaner look. I've changed the file accordingly. Please review and thanks for your guidance to the best possible resolution!